backends/cuda/cuda_partitioner.py

```diff
@@ -54,6 +54,7 @@ def partition(self, exported_program: ExportedProgram) -> PartitionResult:
         partition_tags[tag] = self.delegation_spec
 
         tag_constant_data(exported_program)
+        tag_mutated_buffer(exported_program)
```

Comment:
Make inline comment, something like this

  # Tag mutated buffers (e.g., KV caches) to avoid unnecessary copies.
  # Buffers that are mutated in-place within the delegated subgraph
  # don't need to be returned as outputs.